### DISCUSSION TITLE
Infinite Maurice glitch (Shockjock2) / Beacons damaged ??

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#15172] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Even when triggering the "empty" eventset containing Shockjock2 (also see https://github.com/FakeFishGames/Barotrauma/discussions/14808), beacons get damaged, the NPC drowns, so you do not trigger  the SetDataAction that prevents the event from repeating.

Beacons arent supposed to take damage if the empty eventset is selected, so something fishy is happening
![image](https://github.com/user-attachments/assets/894f07f8-d6ba-43fc-b7cb-5ec768c94bca)

Shockjock2 doesnt power the beacon either, so the pumps don't work and neither does the O2
O Kelly tries to weld it, but since water never goes down he eventually dies

As a result you get infinite Maurices and dead O'Kellies

![image](https://github.com/user-attachments/assets/78ee078a-c7e8-49f4-b1a5-c69a2cfffc3e)
![image](https://github.com/user-attachments/assets/4e11557d-f825-434d-9ba7-34045e5a9e2e)
![image](https://github.com/user-attachments/assets/f81a2506-8f1b-4cff-a4f6-a582f383414d)

### Reproduction steps

Load [Save 65.zip](https://github.com/user-attachments/files/17788491/Save.65.zip)
Teleportsub near beacon
NPC spawns and potentially dies from flooding / unpowered beacon

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

v1.6.19.1 (Unto the Breach Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by A2ur3** [comment]
we also have infinite Rudy

**Comment by Regalis11** [comment]
> Beacons arent supposed to take damage if the empty eventset is selected

[citation needed]

**Reply by ubertpendragon** [reply]
They're in different eventsets... assuming DamageBeaconStationAction is what is damages them, and you didnt leave it hardcoded

**Reply by ubertpendragon** [reply]
https://github.com/FakeFishGames/Barotrauma/blob/7d5b7a310a52372dac0a8bac90fd2a6a2ab342dc/Barotrauma/BarotraumaShared/SharedSource/Map/Levels/Level.cs#L4810-L4812

**Reply by Regalis11** [reply]
As you can see in the code you linked, the walls are damaged if "AllowDamagedWalls" is enabled on the beacon station, which is the case with this specific station.

**Reply by ubertpendragon** [reply]
But this kills the possibility of having events in the beacon because the NPC drowns...

"Empty" eventset should have a sub-eventset that is either
* Don't damage beacon, spawn events
* Damage beacon and nothing else

```xml
      <EventSet identifier="empty" chooserandom="true" triggereventcooldown="false" allowatstart="true" commonness="50">
        <!-- eventset for beacon events, pick events using event TAG so you can easily add beacon events (not a feature?) -->
        <EventSet identifier="beaconevents" chooserandom="false" triggereventcooldown="false" allowatstart="true" commonness="50">
          <ScriptedEvent eventtag="beaconevent" />        
        </EventSet>
        <!-- eventset for truly empty beacon, damages the beacon -->
        <EventSet identifier="emptydamagedbeacon" chooserandom="false" triggereventcooldown="false" allowatstart="true" commonness="50">
          <ScriptedEvent identifier="damagedbeacon">
            <DamageBeaconStationAction DisconnectWireProbability="0" DamageWallProbability="0.25" DamageDeviceProbability="0.5" />
          </ScriptedEvent>      
        </EventSet>
      </EventSet>
```

**Comment by Jasontti** [comment]
Tested a fix for the pre damaged stations.
No issues found and infinite Maurice was fixed earlier.
Closing discussion as completed.
